### PR TITLE
Fix disable some actions when no libraries are open #19129

### PR DESCRIPTION
**Pull request by lzmmxh**
This PR improves the user interface by disabling certain menu items and toolbar buttons when no library is open.
 Specifically:
"Save All" menu item in MainMenu.java is now disabled (greyed out) when there are no open libraries.
"New entry from plain text" menu item in MainMenu.java is now disabled when there are no open libraries.
"Global Search" bar in the main toolbar (MainToolBar.java) is now disabled when there are no open libraries.
"New entry from plain text" button in the main toolbar (MainToolBar.java) is now disabled when there are no open libraries.

## Here's the demo video:
https://github.com/user-attachments/assets/9fb688e5-5dbb-4e20-9f16-b0bf33f691d3

Closes #19129

https://github.com/JabRef/jabref/issues/11923

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/frame/MainToolBar.java
Comment (koppor) [inline src/main/java/org/jabref/gui/frame/MainToolBar.java:110]: Remove empty line - these two statements belong together.

File: src/main/java/org/jabref/gui/frame/MainToolBar.java
Comment (koppor) [inline src/main/java/org/jabref/gui/frame/MainToolBar.java:126]: Do not add spaces. --> Place the cursor on a line with text in IntelliJ and then save.

File: src/main/java/org/jabref/gui/frame/MainToolBar.java
Comment (koppor) [inline src/main/java/org/jabref/gui/frame/MainToolBar.java:131]: No need for the comment - it is self explainiing

```suggestion
                        newEntryFromPlainTextButton,
```

(Did you use ChatGPT? This comment is AI generated, isn't it?)

File: src/main/java/org/jabref/gui/frame/MainMenu.java
Comment (koppor) [inline src/main/java/org/jabref/gui/frame/MainMenu.java:154]: Reorder:

Variable declaration and binding should be grouped.

Now you have

- A-init
- B-init
- empty line
- A-binding
- B-binding

It shoudl be

- A-init
- A-binding
- empty line
- B-init
- B-binding

**Review by koppor — CHANGES_REQUESTED** [review]
Besides minor comments, LGTM :+1:. The changes need a second review and thus this PR awaits a second review.

Please also add a CHANGELOG.md entry

File: src/main/java/org/jabref/gui/frame/MainMenu.java
Comment (lzmmxh) [reply src/main/java/org/jabref/gui/frame/MainMenu.java:154]: ok, I will change it.

**Comment by Siedlerchr** [comment]
please do not delete and re-create your PR! justpush your changes! PR will be auto updted

**Comment by koppor** [comment]
Follow up PR is https://github.com/JabRef/jabref/pull/11936

**Pull request closed without merging**
